### PR TITLE
Split estimate output into separate file and clean up graphs

### PR DESCRIPTION
**Pull request by karlb**
This finishes https://github.com/realrate/RealRate-Private/issues/171 for this repo.

As part of the graphs cleanup, the values for idx, edx and fdx have been removed from output. These should not be necessary now that we distinguish between zeros and NaNs.

### CODE REVIEW

File: causing/examples/__main__.py [hunk 1]
```diff
@@ -49,12 +47,14 @@
 )
 
 # Print json output
-with open(output_dir / "graphs.json", "w") as f:
-    json.dump(graphs, f, sort_keys=True, indent=4)
+dump_json(round_sig_recursive(graphs, 6), output_dir / "graphs.json")
+estimate_dat = estimate.filter_important_keys(estimate_dat)
```
Comment (Holger-Bartel) [inline causing/examples/__main__.py:51]: So are there other data from estimate_dat we do not need? Do we just need the filtered ones?

File: output/example3/graphs.json [hunk 2]
```diff
@@ -308,40 +280,6 @@
     "exj_theo": [
         1.0
     ],
-    "ey_hat": [
-        [
-            1.0,
-            NaN,
```
Comment (Holger-Bartel) [inline output/example3/graphs.json:314]: It's difficult to check if the NaN are right because I can't easily see the old values..

**Review by Holger-Bartel — APPROVED** [review]

File: causing/examples/__main__.py (hunk 1 above)
Comment (karlb) [reply causing/examples/__main__.py:51]: There's also `biases`, `biases_std`, `cov_direct_hat`, `direct_hat`, `hessian_hat` and `sse_hat`. I was unsure whether they can be discarded generally, so I only filtered them before saving the JSON output.

File: output/example3/graphs.json (hunk 2 above)
Comment (karlb) [reply output/example3/graphs.json:314]: These are the old values. Do you mean the new values? Those can be seen in the `estimate.json`.

File: causing/examples/__main__.py (hunk 1 above)
Comment (Holger-Bartel) [reply causing/examples/__main__.py:51]: You're right we need them when evaluting the estimation results to potentially adapt the model.

File: output/example3/graphs.json (hunk 2 above)
Comment (Holger-Bartel) [reply output/example3/graphs.json:314]: Normally, we have a line to line comparison (+/-); it is just difficult to read the changes from here.

**Pull request merged**